File: 89.gray-code.cpp
class Solution {
public:
    vector<int> grayCode(int n) {
        if (n == 0) {
            return {0};
        }
        auto ans = grayCode(n - 1);
        int s = ans.size();
        for (int i = s - 1; i >= 0; i--) {
            ans.push_back(s + ans[i]);
        }
        return ans;
    }
};

File: 89.gray-code.py
class Solution:
    def grayCode(self, n):
        """
        :type n: int
        :rtype: List[int]
        """
        if n == 0:
            return [0]
        ans = self.grayCode(n - 1)
        s = len(ans)
        for i in range(s - 1, -1, -1):
            ans.append(s + ans[i])
        return ans